File: src/test/java/com/codingbat/warmup1/SumDoubleTest.java
package com.codingbat.warmup1;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class SumDoubleTest {
  @Test
  public void shouldEqualThreeForOneTwoSumDouble() {
    assertEquals(3, SumDouble.sumDouble(1, 2));
  }

  @Test
  public void shouldEqualFiveForThreeTwoSumDouble() {
    assertEquals(5, SumDouble.sumDouble(3, 2));
  }

  @Test
  public void shouldEqualEightForTwoTwoSumDouble() {
    assertEquals(8, SumDouble.sumDouble(2, 2));
  }

  @Test
  public void shouldEqualNegativeOneForNegativeOneZeroSumDouble() {
    assertEquals(-1, SumDouble.sumDouble(-1, 0));
  }

  @Test
  public void shouldEqualTwelveForThreeThreeSumDouble() {
    assertEquals(12, SumDouble.sumDouble(3, 3));
  }

  @Test
  public void shouldEqualZeroForZeroZeroSumDouble() {
    assertEquals(0, SumDouble.sumDouble(0, 0));
  }

  @Test
  public void shouldEqualOneForZeroOneSumDouble() {
    assertEquals(1, SumDouble.sumDouble(0, 1));
  }

  @Test
  public void shouldEqualSevenForThreeFourSumDouble() {
    assertEquals(7, SumDouble.sumDouble(3, 4));
  }
}

File: src/test/kotlin/com/codingbat/warmup1/SumDoubleTest.kt
package com.codingbat.warmup1

import org.junit.Test
import kotlin.test.assertEquals

class SumDoubleTestKt {
  @Test
  fun shouldEqualThreeForOneTwoSumDouble() {
    assertEquals(3, sumDouble(1, 2).toLong())
  }

  @Test
  fun shouldEqualFiveForThreeTwoSumDouble() {
    assertEquals(5, sumDouble(3, 2).toLong())
  }

  @Test
  fun shouldEqualEightForTwoTwoSumDouble() {
    assertEquals(8, sumDouble(2, 2).toLong())
  }

  @Test
  fun shouldEqualNegativeOneForNegativeOneZeroSumDouble() {
    assertEquals(-1, sumDouble(-1, 0).toLong())
  }

  @Test
  fun shouldEqualTwelveForThreeThreeSumDouble() {
    assertEquals(12, sumDouble(3, 3).toLong())
  }

  @Test
  fun shouldEqualZeroForZeroZeroSumDouble() {
    assertEquals(0, sumDouble(0, 0).toLong())
  }

  @Test
  fun shouldEqualOneForZeroOneSumDouble() {
    assertEquals(1, sumDouble(0, 1).toLong())
  }

  @Test
  fun shouldEqualSevenForThreeFourSumDouble() {
    assertEquals(7, sumDouble(3, 4).toLong())
  }
}
